httomo/runner/dataset.py

```diff
@@ -41,28 +79,50 @@ def __init__(
         chunk_index[slicing_dim] += block_start
         self._chunk_index = make_3d_shape_from_shape(chunk_index)
         global_index = [0, 0, 0]
-        global_index[slicing_dim] += chunk_start + block_start
+        global_index[slicing_dim] += chunk_start + block_start + padding[0]
         self._global_index = make_3d_shape_from_shape(global_index)
 
         self._check_inconsistencies()
-        
+
     def _check_inconsistencies(self):
-        if self.chunk_index[self.slicing_dim] < 0:
+        if self.chunk_index[self.slicing_dim] + self._padding[0] < 0:
             raise ValueError("block start index must be >= 0")
-        if self.chunk_index[self.slicing_dim] + self.shape[self.slicing_dim] > self.chunk_shape[self.slicing_dim]:
+        if (
+            self.chunk_index[self.slicing_dim]
+            + self.shape[self.slicing_dim]
+            - self._padding[1]
+            > self.chunk_shape[self.slicing_dim]
+        ):
             raise ValueError("block spans beyond the chunk's boundaries")
-        if self.global_index[self.slicing_dim] < 0:
+        if self.global_index[self.slicing_dim] + self._padding[0] < 0:
             raise ValueError("chunk start index must be >= 0")
-        if self.global_index[self.slicing_dim] + self.shape[self.slicing_dim] > self.global_shape[self.slicing_dim]:
+        if (
+            self.global_index[self.slicing_dim]
+            + self.shape[self.slicing_dim]
+            - self._padding[1]
+            > self.global_shape[self.slicing_dim]
+        ):
             raise ValueError("chunk spans beyond the global data boundaries")
-        if any(self.chunk_shape[i] > self.global_shape[i] for i in range(3)):    
+        if any(self.chunk_shape[i] > self.global_shape[i] for i in range(3) if i != self.slicing_dim):
+            raise ValueError("chunk shape is larger than the global shape")
+        if self.chunk_shape[self.slicing_dim] - self.padding[0] - self.padding[1] > self.global_shape[self.slicing_dim]:
```

Comment:
After some thinking, I _think_ I understand why the second `if` condition has been added here (to catch an _unpadded_ chunk that is larger than the global shape in the slicing dim?).

However, both checks produce the exact same error message of "chunk shape is larger than the global shape": even if it's technically a correct error message for both cases, I'd prefer there to be some distinction between the two message, so then someone reading the code would have a better chance of understanding the difference between the two checks.

Are the two checks something like:
- first check: "is the chunk (for both padded and unpadded cases) larger than the global data in the non-slicing dims?"
- second check: "is the chunk (for both padded and unpadded) larger than the global data in the slicing dim?"

(Maybe not quite, I'm not sure)

Reply:
fixed in fddde0d05